luatest/server.lua

```diff
@@ -36,47 +49,109 @@ local Server = {
+--   value of the `TARANTOOL_WORKDIR` env variable which is passed into the
+--   server process.
+--   Defaults to `<vardir>/<alias>-<random id>`.
+-- @string[opt] object.datadir Directory path whose contents will be recursively
+--   copied into `object.workdir` during initialization.
+-- @number[opt] object.http_port Port for HTTP connection to the new server and
+--   the value of the `TARANTOOL_HTTP_PORT` env variable which is passed into
+--   the server process.
+--   Not supported in the default `server_instance.lua` script.
+-- @number[opt] object.net_box_port Port for the `net.box` connection to the new
+--   server and the value of the `TARANTOOL_LISTEN` env variable which is passed
+--   into the server process.
+-- @string[opt] object.net_box_uri URI for the `net.box` connection to the new
+--   server and the value of the `TARANTOOL_LISTEN` env variable which is passed
+--   into the server process.
+-- @tab[opt] object.net_box_credentials Override the default credentials for the
+--   `net.box` connection to the new server.
+-- @tab[opt] object.box_cfg Extra options for `box.cfg()` and the value of the
+--   `TARANTOOL_BOX_CFG` env variable which is passed into the server process.
+-- @return table
 function Server:new(object)
     checks('table', self.constructor_checks)
     self:inherit(object)
     object:initialize()
     return object
 end
 
+-- Initialize the server object.
 function Server:initialize()
+    if self.id == nil then
+        self.id = digest.base64_encode(digest.urandom(9), {urlsafe = true})
+    end
+
+    if self.alias == nil then
+        self.alias = DEFAULT_ALIAS
+    end
+
+    if self.command == nil then
+        self.command = DEFAULT_INSTANCE
+    end
+
+    if self.workdir == nil then
+        self.workdir = fio.pathjoin(self.vardir, ('%s-%s'):format(self.alias, self.id))
+        fio.rmtree(self.workdir)
+        fio.mktree(self.workdir)
+    end
+
+    if self.datadir ~= nil then
+        local ok, err = fio.copytree(self.datadir, self.workdir)
+        if not ok then
+            error(('Failed to copy directory: %s'):format(err))
+        end
+        self.datadir = nil
+    end
+
     if self.http_port then
         self.http_client = http_client.new()
     end
-    if self.net_box_uri then
-        local system = os.execute('[ $(uname) = Linux ]') == 0 and 'linux' or 'other'
-        if string.len(self.net_box_uri) > max_unix_socket_path[system] then
-            error(string.format('Net box URI must be <= max Unix domain socket path length (%s chars)',
-                max_unix_socket_path[system]))
-        end
+
+    if self.net_box_uri == nil and self.net_box_port == nil then
+        self.net_box_uri = self.build_listen_uri(self.alias)
+        fio.mktree(self.vardir)
     end
     if self.net_box_uri == nil and self.net_box_port then
         self.net_box_uri = 'localhost:' .. self.net_box_port
     end
+    if uri.parse(self.net_box_uri).host == 'unix/' then
```

Comment:
when is it not unix?
when is it unix, but not linux? with macos?

Reply:
It is about UNIX sockets, not operating system name.